### ISSUE TITLE
DS - Snackbar - Improve screen reader behavior

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/va-mobile-library#440]
## Description 
There was [a slack thread discussion](https://adhoc.slack.com/archives/C05DQPHUVNZ/p1723231477940539) on whether the behavior called for by the Figma doc for Snackbar:
> "Snackbar or toast must be dynamically announced by the screen reader without moving focus to it."
Current: App automatically moves the focus to the snackbar.
Proposed: Snackbar is announced, but focus is not moved.

could be further improved for blind/low-vision (screen reader) users. For example: auto-dismissing it when there's no action to take and it's purely informational.

## Acceptance Criteria
  

-  [ ] Determine if further behavior refinement is desirable (e.g. auto-dismissing the Snackbar after 5 seconds if there's no follow-up action to Undo/Try again) and, if so, implement it

**Accessibility Requirements**
- [ ] Behavior enhanced according to specifications worked out with Brea
